### PR TITLE
Create data models

### PR DESCRIPTION
**Pull request by MandlaRetro**
Review the data Models

### CODE REVIEW

File: SkietbaanBE/SkietbaanBE/Migrations/20190206075919_MagicMigration.cs [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+using System;
+using System.Collections.Generic;
+
+namespace SkietbaanBE.Migrations
+{
+    public partial class MagicMigration : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+
```
Comment (ash-rabbit) [inline SkietbaanBE/SkietbaanBE/Migrations/20190206075919_MagicMigration.cs]: Please remove this whitespace

File: SkietbaanBE/SkietbaanBE/Migrations/ModelsContextModelSnapshot.cs [hunk 2]
```diff
@@ -57,6 +93,81 @@ protected override void BuildModel(ModelBuilder modelBuilder)
 
                     b.ToTable("Users");
                 });
+
+            modelBuilder.Entity("SkietbaanBE.Models.UserCompStats", b =>
+                {
+                    b.Property<int>("Id")
+                        .ValueGeneratedOnAdd();
+
```
Comment (BWMuller) [inline SkietbaanBE/SkietbaanBE/Migrations/ModelsContextModelSnapshot.cs]: No need for all this white space. Same with the above changes

File: SkietbaanBE/SkietbaanBE/Program.cs [hunk 3]
```diff
@@ -14,12 +14,13 @@ public class Program
     {
         public static void Main(string[] args)
         {
-            BuildWebHost(args).Run();
+            Console.Write(Environment.GetEnvironmentVariable("TEMP"));
+           // BuildWebHost(args).Run();
```
Comment (FredJEhlers) [inline SkietbaanBE/SkietbaanBE/Program.cs]: Remove Ghost code please in all instances. Also the additional lines between code please clean that up as well

File: SkietbaanBE/SkietbaanBE/Migrations/20190206075919_MagicMigration.cs [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+                        principalColumn: "Id",
+                        onDelete: ReferentialAction.Restrict);
+                });
+
+            migrationBuilder.CreateTable(
+                name: "UserCompStats",
+                columns: table => new
+                {
+                    Id = table.Column<int>(nullable: false)
+                        .Annotation("SqlServer:ValueGenerationStrategy", SqlServerValueGenerationStrategy.IdentityColumn),
```
Comment (CilliersN) [inline SkietbaanBE/SkietbaanBE/Migrations/20190206075919_MagicMigration.cs]: SqlServerValueGenerationStrategy.IdentityColumn) on a new line

File: SkietbaanBE/SkietbaanBE/Models/UserCompStats.cs [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+namespace SkietbaanBE.Models {
+    public class UserCompStats {
+        [Key]
+        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+        public int Id { get; set; }
+
+        public virtual User User { get; set; }
+
+        public virtual Competition Competition { get; set; }
```
Comment (grootendorste) [inline SkietbaanBE/SkietbaanBE/Models/UserCompStats.cs]: You can maybe group the relevant parts with each other.

File: SkietbaanBE/SkietbaanBE/Program.cs [hunk 6]
```diff
@@ -14,12 +14,13 @@ public class Program
...
         }
 
-        public static IWebHost BuildWebHost(string[] args) =>
+        /*public static IWebHost BuildWebHost(string[] args) =>
             WebHost.CreateDefaultBuilder(args)
                 .UseStartup<Startup>()
-                .Build();
+                .Build();*/
```
Comment (BWMuller) [inline SkietbaanBE/SkietbaanBE/Program.cs]: GC

File: SkietbaanBE/SkietbaanBE/Migrations/20190206075919_MagicMigration.cs (hunk 4 above)
Comment (CilliersN) [inline SkietbaanBE/SkietbaanBE/Migrations/20190206075919_MagicMigration.cs]: indentation

File: SkietbaanBE/SkietbaanBE/Program.cs [hunk 7]
```diff
@@ -14,12 +14,13 @@ public class Program
     {
         public static void Main(string[] args)
         {
-            BuildWebHost(args).Run();
+            Console.Write(Environment.GetEnvironmentVariable("TEMP"));
+           // BuildWebHost(args).Run();
         }
 
-        public static IWebHost BuildWebHost(string[] args) =>
+        /*public static IWebHost BuildWebHost(string[] args) =>
```
Comment (CilliersN) [inline SkietbaanBE/SkietbaanBE/Program.cs]: remove GC or add TODO

Comment (ash-rabbit) [inline SkietbaanBE/SkietbaanBE/Program.cs]: I explained about having ghost code in the forum. Please follow the instructions

File: SkietbaanBE/SkietbaanBE/Program.cs [hunk 8]
```diff
@@ -14,12 +14,13 @@ public class Program
     {
         public static void Main(string[] args)
         {
-            BuildWebHost(args).Run();
+            Console.Write(Environment.GetEnvironmentVariable("TEMP"));
```
Comment (grootendorste) [inline SkietbaanBE/SkietbaanBE/Program.cs]: Is TEMP for testing purposes?

Comment (grootendorste) [reply SkietbaanBE/SkietbaanBE/Program.cs]: Also, shouldn't the consoles be taken out?

Comment (MandlaRetro) [reply SkietbaanBE/SkietbaanBE/Program.cs]: Ja I wanted to show someone how we could use the environment variable to store the connection string. Then i forgot to remove it

File: SkietbaanBE/SkietbaanBE/Migrations/20190206075919_MagicMigration.cs (hunk 1 above)
Comment (Lavhe) [reply SkietbaanBE/SkietbaanBE/Migrations/20190206075919_MagicMigration.cs]: @ash-rabbit thats an autogenerated file 😆

File: SkietbaanBE/SkietbaanBE/Models/Group.cs [hunk 9]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SkietbaanBE.Models {
+    public class Group {
+        [Key]
```
Comment (Icecoder1) [inline SkietbaanBE/SkietbaanBE/Models/Group.cs]: If you called the int Id GroupId Entity Framework to know its the primary key.

**Pull request closed without merging**